analysis/src/xml.rs

```diff
@@ -0,0 +1,715 @@
+            c_declaration: descendant_text(node),
+            values: attribute(node, "values"),
+            len: attribute(node, "len"),
+            altlen: attribute(node, "altlen"),
+            optional: attribute(node, "optional"),
+        }
+    }
+}
+
+#[derive(Debug)]
+pub struct Structure {
+    pub name: XmlStr,
+    pub structextends: Option<XmlStr>,
+    pub members: Vec<StructureMember>,
+}
+
+impl Structure {
+    fn from_node(node: Node, api: &str) -> Structure {
+        Structure {
+            name: attribute(node, "name").unwrap(),
+            structextends: attribute(node, "structextends"),
+            members: node
+                .children()
+                .filter(|node| node.has_tag_name("member"))
+                .filter(|node| api_matches(node, api))
+                .map(StructureMember::from_node)
+                .collect(),
+        }
+    }
+}
+
+#[derive(Debug)]
+pub struct Constant {
+    pub ty: XmlStr,
+    pub value: XmlStr,
+    pub name: XmlStr,
+}
+
+impl Constant {
+    fn from_node(node: Node) -> Constant {
+        Constant {
+            ty: attribute(node, "type").unwrap(),
+            value: attribute(node, "value").unwrap(),
+            name: attribute(node, "name").unwrap(),
+        }
+    }
+}
+
+#[derive(Debug)]
+pub struct EnumValue {
+    pub value: XmlStr,
+    pub name: XmlStr,
+}
+
+impl EnumValue {
+    fn from_node(node: Node) -> EnumValue {
+        EnumValue {
+            value: attribute(node, "value").unwrap(),
+            name: attribute(node, "name").unwrap(),
+        }
+    }
+}
+
+#[derive(Debug)]
+pub struct Enum {
+    pub name: XmlStr,
+    pub values: Vec<EnumValue>,
+    pub aliases: Vec<Alias>,
+}
+
+impl Enum {
+    fn from_node(node: Node, api: &str) -> Enum {
+        let mut value = Enum {
+            name: attribute(node, "name").unwrap(),
+            values: Vec::new(),
+            aliases: Vec::new(),
+        };
+
+        for variant in node
+            .children()
+            .filter(|node| node.has_tag_name("enum"))
+            .filter(|node| api_matches(node, api))
+        {
+            if variant.has_attribute("alias") {
+                value.aliases.push(Alias::from_node(variant));
+            } else {
+                value.values.push(EnumValue::from_node(variant));
+            }
+        }
+
+        value
+    }
+}
+
+#[derive(Debug)]
+pub struct BitMaskBit {
+    pub bitpos: XmlStr,
+    pub name: XmlStr,
+}
+
+impl BitMaskBit {
+    fn from_node(node: Node) -> BitMaskBit {
+        BitMaskBit {
+            bitpos: attribute(node, "bitpos").unwrap(),
+            name: attribute(node, "name").unwrap(),
+        }
+    }
+}
+
+#[derive(Debug)]
+pub struct BitMask {
+    pub name: XmlStr,
+    pub bits: Vec<BitMaskBit>,
+    /// Some bitmask variants represent literal values instead of specific
+    /// individual bits, e.g. a combination of bits, or no bits at all. An good
+    /// example for this is in `VkCullModeFlagBits`.
```

Comment:
```suggestion
    /// example for this is `VkCullModeFlagBits`.
```

Perhaps point to `FRONT_AND_BACK`?